pgjdbc/src/main/java/org/postgresql/jdbc/PgArray.java

```diff
@@ -185,7 +187,8 @@ public Object getArrayImpl(long index, int count, Map<String, Class<?>> map) thr
 
   private Object readBinaryArray(int index, int count) throws SQLException {
     int dimensions = ByteConverter.int4(fieldBytes, 0);
-    // int flags = ByteConverter.int4(fieldBytes, 4); // bit 0: 0=no-nulls, 1=has-nulls
+    // int flags = ByteConverter.int4(fieldBytes, 4); // bit 0: 0=no-nulls,
+    // 1=has-nulls
     int elementOid = ByteConverter.int4(fieldBytes, 8);
     int pos = 12;
```

Comment:
@bokken can you do me a favour and document the postgres array layout so that we know why you are moving forward 12 bytes. I have looked at this before and forgotten why this is so. It would be a huge time savings to put the layout in this file and possibly reference the code in the pg source where you found it.

Reply:
@davecramer, I reversed engineered what PGArray is doing when processing a binary array in response. I could not find the pg source where this is being done

Reply:
src/util/adt/arrayfuncs.c is where you will find all the code. Not so much documentation

Reply:
awesome!